django/db/models/expressions.py

```diff
@@ -104,6 +104,8 @@ def bitrightshift(self, other):
         return self._combine(other, self.BITRIGHTSHIFT, False)
 
     def __or__(self, other):
+        if getattr(self, 'conditional', False) and getattr(other, 'conditional', False):
+            return Q(self) | Q(other)
```

Comment:
Um, so if we're adding this here, should we not also add the following to `__and__()`?
```python
        if getattr(self, 'conditional', False) and getattr(other, 'conditional', False):
            return Q(self) & Q(other)
```
Or am I missing something?